pkg/reconciler/broker/broker.go

```diff
@@ -126,6 +132,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, key string) error {
 }
 
 func (r *Reconciler) reconcile(ctx context.Context, b *v1alpha1.Broker) error {
+	if b.Spec.ChannelTemplate.Kind != "" && b.Spec.ChannelTemplate.APIVersion != "" {
+		return r.reconcileCRD(ctx, b)
```

Comment:
I wonder if a future pattern should be to add a new reconciler + this filter. It would make deleting the legacy parts easier.

Reply:
hard to know if it really would help or not...